test/conformance/tests/sriov_operator.go

```diff
@@ -681,292 +423,566 @@ var _ = Describe("[sriov] operator", func() {
+						resNum, _ := testedNode.Status.Allocatable["openshift.io/apivolresource"]
+						capacity, _ := resNum.AsInt64()
+						return capacity
+					}, 3*time.Minute, time.Second).Should(Equal(int64(5)))
+
+					sriovNetwork := &sriovv1.SriovNetwork{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "test-apivolnetwork",
+							Namespace: operatorNamespace,
+						},
+						Spec: sriovv1.SriovNetworkSpec{
+							ResourceName:     "apivolresource",
+							IPAM:             `{"type":"host-local","subnet":"10.10.10.0/24","rangeStart":"10.10.10.171","rangeEnd":"10.10.10.181","routes":[{"dst":"0.0.0.0/0"}],"gateway":"10.10.10.1"}`,
+							NetworkNamespace: namespaces.Test,
+						}}
+					err = clients.Create(context.Background(), sriovNetwork)
 					Expect(err).ToNot(HaveOccurred())
+
 					Eventually(func() error {
 						netAttDef := &netattdefv1.NetworkAttachmentDefinition{}
-						return clients.Get(context.Background(), runtimeclient.ObjectKey{Name: ipv6NetworkName, Namespace: namespaces.Test}, netAttDef)
+						return clients.Get(context.Background(), runtimeclient.ObjectKey{Name: "test-apivolnetwork", Namespace: namespaces.Test}, netAttDef)
 					}, 10*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
 
-					pod := createTestPod(testNode, []string{ipv6NetworkName})
-					ips, err := network.GetSriovNicIPs(pod, "net1")
+					podDefinition := pod.DefineWithNetworks([]string{sriovNetwork.Name})
+					created, err := clients.Pods(namespaces.Test).Create(podDefinition)
 					Expect(err).ToNot(HaveOccurred())
-					Expect(ips).NotTo(BeNil(), "No sriov network interface found.")
-					Expect(len(ips)).Should(Equal(1))
-					for _, ip := range ips {
-						pingPod(ip, testNode, ipv6NetworkName)
+
+					var runningPod *corev1.Pod
+					Eventually(func() corev1.PodPhase {
+						runningPod, err = clients.Pods(namespaces.Test).Get(created.Name, metav1.GetOptions{})
+						Expect(err).ToNot(HaveOccurred())
+						return runningPod.Status.Phase
+					}, 3*time.Minute, time.Second).Should(Equal(corev1.PodRunning))
+
+					var downwardVolume *corev1.Volume
+					for _, v := range runningPod.Spec.Volumes {
+						if v.Name == "podnetinfo" {
+							downwardVolume = v.DeepCopy()
+							break
+						}
 					}
+
+					Expect(downwardVolume).ToNot(BeNil(), "Downward volume not found")
+					Expect(downwardVolume.DownwardAPI).ToNot(BeNil(), "Downward api not found in volume")
+					Expect(downwardVolume.DownwardAPI.Items).To(SatisfyAll(
+						ContainElement(corev1.DownwardAPIVolumeFile{
+							Path: "labels",
+							FieldRef: &corev1.ObjectFieldSelector{
+								APIVersion: "v1",
+								FieldPath:  "metadata.labels",
+							},
+						}), ContainElement(corev1.DownwardAPIVolumeFile{
+							Path: "annotations",
+							FieldRef: &corev1.ObjectFieldSelector{
+								APIVersion: "v1",
+								FieldPath:  "metadata.annotations",
+							},
+						})))
+				})
+				Context("Multiple sriov device and attachment", func() {
```

Comment:
This is nested in `Context("Resource Injector", func() { ` but I think it should be on the same level (it was already wrong before). Also, I think this does not depend on a specific policy either.

Reply:
Done